src/driver/amdxdna/aie2_smu.c

```diff
@@ -218,10 +218,12 @@ int aie2_smu_start(struct amdxdna_dev_hdl *ndev)
 	struct smu *smu = &ndev->smu;
 	int ret;
 
-	ret = aie2_smu_set_power_on(ndev);
-	if (ret) {
-		XDNA_ERR(ndev->xdna, "Power on failed, ret %d", ret);
-		return ret;
+	if (ndev->smu.power_state == SMU_POWER_OFF) {
```

Comment:
Can we move the check into aie2_smu_set_power_on()? Is there a reason we need to expose it to caller outside of aie2_smu_set_power_on/_off()?

Reply:
I guess that makes sense, I just didn't want to call the power on again in the initial flow, I think your suggestion makes sense too. I'll update this.

Reply:
I understand what you want to do and moving the check into aie2_smu_set_power_on() does not change the logic. Then, we can keep smu.power_state internal to  aie2_smu_set_power_on()/_off() so it'll become a cleaner solution.